src/EoS/Nuclear/NuclearEoS.cpp

```diff
@@ -23,122 +40,213 @@ void findenergy( const real x, const real y, const real z,
+//                keyerr          : output error
+//                                  667 : fail in finding T/e (temp/energy-based table)
+//                                  101 : Y_e too high
+//                                  102 : Y_e too low
+//                                  103 : temp too high 
+//                                  104 : temp too low
+//                                  105 : rho too high
+//                                  106 : rho too low
+//                                  107 : eps too high     (energy-based table)
+//                                  108 : eps too low      (energy-based table)
+//                                  109 : temp too high    (energy-based table)
+//                                  110 : temp too low     (energy-based table)
+//                                  111 : entropy too high (energy-based table)
+//                                  112 : entropy too low  (energy-based table)
+//                                  113 : log(P) too high  (energy-based table)
+//                                  114 : log(P) too low   (energy-based table)
+//                rfeps           : tolerence for interpolations
 //-----------------------------------------------------------------------------------------------
 GPU_DEVICE
-void nuc_eos_C_short( const real xrho, real *xenr, const real xye,
-                      real *xtemp, real *xent, real *xprs,
+void nuc_eos_C_short( const real xrho, real *xtemp, const real xye,
+                      real *xenr, real *xent, real *xprs,
                       real *xcs2, real *xmunu, const real energy_shift,
-                      const int nrho, const int neps, const int nye, const int nmode,
+                      const int nrho, const int ntoreps, const int nye,
+                      const int nrho_mode, const int nmode, const int nye_mode,
                       const real *alltables, const real *alltables_mode,
-                      const real *logrho, const real *logeps, const real *yes,
-                      const real *logtemp_mode, const real *entr_mode, const real *logprss_mode,
+                      const real *logrho, const real *logtoreps, const real *yes, const real *logrho_mode,
+                      const real *logepsort_mode, const real *entr_mode, const real *logprss_mode, const real *yes_mode,
                       const int keymode, int *keyerr, const real rfeps )
 {
 
 // check whether the input density and Ye are within the table
    const real lr = LOG10( xrho );
    *keyerr = 0;
 
+
    if ( lr > logrho[nrho-1] )  {  *keyerr = 105;  return;  }
    if ( lr < logrho[     0] )  {  *keyerr = 106;  return;  }
-   if ( lr != lr )             {  *keyerr = 203;  return;  }
 
    if ( xye > yes  [nye -1] )  {  *keyerr = 101;  return;  }
    if ( xye < yes  [     0] )  {  *keyerr = 102;  return;  }
-   if ( xye != xye )           {  *keyerr = 201;  return;  }
```

Comment:
Why remove the sanity check of NaN for density and ye, and checks in other places?

Reply:
I added back.